### PR TITLE
Exists method added in BigtableInstanceAdminClient

### PR DESCRIPTION
**Pull request by elisheva-qlogic**

### CODE REVIEW

**Comment by sduskis** [comment]
@elisheva-qlogic, there's a compilation error:

> [ERROR] COMPILATION ERROR :
> [INFO] -------------------------------------------------------------
> [ERROR] /tmpfs/src/github/google-cloud-java/google-cloud-clients/google-cloud-bigtable-admin/src/test/java/com/google/cloud/bigtable/admin/v2/BigtableInstanceAdminClientTest.java:[873,9] no suitable method found for thenReturn(com.google.api.core.ApiFuture<java.lang.Object>)
>     method org.mockito.stubbing.OngoingStubbing.thenReturn(com.google.api.core.ApiFuture<com.google.bigtable.admin.v2.Instance>) is not applicable
>       (argument mismatch; com.google.api.core.ApiFuture<java.lang.Object> cannot be converted to com.google.api.core.ApiFuture<com.google.bigtable.admin.v2.Instance>)
>     method org.mockito.stubbing.OngoingStubbing.thenReturn(com.google.api.core.ApiFuture<com.google.bigtable.admin.v2.Instance>,com.google.api.core.ApiFuture<com.google.bigtable.admin.v2.Instance>...) is not applicable
>       (argument mismatch; com.google.api.core.ApiFuture<java.lang.Object> cannot be converted to com.google.api.core.ApiFuture<com.google.bigtable.admin.v2.Instance>)
> [INFO] 1 error
> [INFO] -------------------------------------------------------------
> [INFO] ------------------------------------------------------------------------
>

**Comment by igorbernstein2** [comment]
I think I just fixed the build, but I'm not sure why the CLA is complaining

**Comment by chingor13** [comment]
@igorbernstein2 The merge commit 8e6d2fab2da81bd7fa1a19d4899ac663a3a6d250 looks like it was done on your macbook without git configured for your email address. The CLA bot can't find a signed CLA for that email address.

**Comment by chingor13** [comment]
IT failure was logging test quota issue

File: google-cloud-clients/google-cloud-bigtable-admin/src/main/java/com/google/cloud/bigtable/admin/v2/BigtableInstanceAdminClient.java [hunk 1]
```diff
@@ -417,6 +418,68 @@ public Void apply(Empty input) {
...
+   *  },
+   *  MoreExecutors.directExecutor()
+   * );
+   * }</pre>
+   */
+  public ApiFuture<Boolean> existsAsync(String instanceId) {
+    ApiFuture<Instance> protoFuture = getInstanceAsync(instanceId);
+
+    ApiFuture<Boolean> existsFuture = ApiFutures
+        .transform(protoFuture, new ApiFunction<Instance, Boolean>() {
```
Comment (chingor13) [inline google-cloud-clients/google-cloud-bigtable-admin/src/main/java/com/google/cloud/bigtable/admin/v2/BigtableInstanceAdminClient.java]: This ApiFunction could be made into a constant so we don't need to instantiate a new one every call.

File: google-cloud-clients/google-cloud-bigtable-admin/src/main/java/com/google/cloud/bigtable/admin/v2/BigtableInstanceAdminClient.java [hunk 2]
```diff
@@ -417,6 +418,68 @@ public Void apply(Empty input) {
...
+          @Override
+          public Boolean apply(Instance ignored) {
+            return true;
+          }
+        }, MoreExecutors.directExecutor());
+
+    return ApiFutures.catching(existsFuture, NotFoundException.class,
+        new ApiFunction<NotFoundException, Boolean>() {
```
Comment (chingor13) [inline google-cloud-clients/google-cloud-bigtable-admin/src/main/java/com/google/cloud/bigtable/admin/v2/BigtableInstanceAdminClient.java]: ditto on this ApiFunction

File: google-cloud-clients/google-cloud-bigtable-admin/src/main/java/com/google/cloud/bigtable/admin/v2/BigtableInstanceAdminClient.java (hunk 1 above)
Comment (igorbernstein2) [reply google-cloud-clients/google-cloud-bigtable-admin/src/main/java/com/google/cloud/bigtable/admin/v2/BigtableInstanceAdminClient.java]: I'm pretty sure the jvm will optimize away the allocation and I think moving this to a non-local const will decrease readability

Comment (chingor13) [reply google-cloud-clients/google-cloud-bigtable-admin/src/main/java/com/google/cloud/bigtable/admin/v2/BigtableInstanceAdminClient.java]: SGTM

**Pull request merged**